council/llm/providers/anthropic/anthropic_llm_configuration.py

```diff
@@ -124,8 +124,11 @@ def from_env() -> AnthropicLLMConfiguration:
         config._read_optional_env()
         return config
 
-    @staticmethod
-    def from_spec(spec: LLMConfigSpec) -> AnthropicLLMConfiguration:
+    @classmethod
+    def from_spec(cls, spec: LLMConfigSpec) -> AnthropicLLMConfiguration:
+        if not spec.provider.is_of_kind(LLMProviders.Anthropic):
+            raise ValueError(f"Invalid LLM provider, actual {spec.provider}, expected {LLMProviders.Anthropic}")
+
```

Comment:
Add a `check_provider` in the base class or `LLMConfigSpec`

`def check_provider(provider: LLMProviders) -> None:
        if not spec.provider.is_of_kind(provider):
            raise ValueError(f"Invalid LLM provider, actual {spec.provider}, expected {provider}")
`